### PR TITLE
add cn version 'declarative-object-management-configuration.md'

### PR DESCRIPTION
**Pull request by Lion-Wei**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4913)

### CODE REVIEW

**Comment by Lion-Wei** [comment]
@chenopis Sorry to ask, but I don't understand what this failing mean, how should i modify this file?

**Comment by chenopis** [comment]
@Lion-Wei Unfortunately, only admins can see the error messages. :( 

screen shot 2017-08-17 at 12 06 38 pm

It is saying that it the file `simple_deployment.yaml` cannot be found. It is probably coming from the include code.html lines, e.g.

```
{% include code.html language="yaml" file="simple_deployment.yaml" ghlink="/docs/tutorials/object-management-kubectl/simple_deployment.yaml" %}
```

You may want to copy the file to somewhere in the `/cn` directory structure and change the file path of those include tags.

**Comment by chenopis** [comment]
cc @markthink

**Comment by Lion-Wei** [comment]
thx, finally worked.

**Comment by chenopis** [comment]
Once someone from one of the other CN orgs gives an LGTM, I can merge this.

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 1]
```diff
@@ -0,0 +1,886 @@
+---
+title: 使用配置文件以声明的方式管理Kubernetes对象
+---
+
+{% capture overview %}
+将多个对象配置文件存储在文件夹中，使用 `kubectl apply` 可以递归的创建和更新这些对象。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 原文中的“...as needed"很关键，表达了kubectl apply进行三方比较的逻辑，译文中没有体现。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 2]
```diff
@@ -0,0 +1,886 @@
...
+可以通过这种方式进行Kubernetes对象的创建，更新和删除操作。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 中文表达中，尽量避免这种无主句。即使是方便阅读，分割出来的长句，也应该补充主语。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 3]
```diff
@@ -0,0 +1,886 @@
...
+这种方式保留了对活动对象的写入，而不将改动合并到对象配置文件中。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 英文"retain"的翻译太生硬了。
live objects一般译为”活跃对象“

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 4]
```diff
@@ -0,0 +1,886 @@
...
+{% endcapture %}
+
+{% capture body %}
+
+## 权衡
+
+`kubectl` 工具支持三种对象的管理:
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 不是”三种对象的管理“，而是”三种对象管理“

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 5]
```diff
@@ -0,0 +1,886 @@
...
+
+* 命令式的方式
+* 命令式的对象配置
+* 声明式的对象配置
+
+参见[Kubernetes对象管理](/docs/concepts/tools/kubectl/object-management-overview/)
+讨论各种对象管理的优缺点.
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: for a discussion of ... 不是让读者参与讨论，是让读者去阅读关于...的讨论。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 6]
```diff
@@ -0,0 +1,886 @@
...
+
+## 前言
+
+声明式对象管理需要对Kubernetes对象定义和配置有明确的了解。如果您还没有，
+请阅读下面文档：
+
+- [使用命令式的方式管理 Kubernetes 对象](/docs/concepts/tools/kubectl/object-management-using-imperative-commands/)
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 这里的imperative直接译为"命令"就好了，没必要译为”命令式"

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 7]
```diff
@@ -0,0 +1,886 @@
...
+- [使用配置文件的方式管理Kubernetes对象](/docs/concepts/tools/kubectl/object-management-using-imperative-config/)
+
+下面是本文档中使用到的术语定义：
+
+- *对象配置文件 / 配置文件*：一个定义Kubernetes对象配置的文件。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 英文里面可以用斜体表示强调，中文斜体可读性太差，可以考虑使用加粗：

**对象配置文件／配置文件**

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 8]
```diff
@@ -0,0 +1,886 @@
...
+  本主题介绍如何将配置文件传递给 `kubectl apply`。配置文件通常保存在源控件中，
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: “source control"指的是代码版本控制系统，如CVS、SVN、GIT。翻译成”源控件"是错误的。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 9]
```diff
@@ -0,0 +1,886 @@
...
+  比如Git。
+- *活动对象配置 / 活动配置*：由Kubernetes集群观察到的一个对象的实时配置值。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: live -> “活跃”或"当前"，译为“实时”很夸张

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 10]
```diff
@@ -0,0 +1,886 @@
...
+  它被保存在Kubernetes集群存储中，通常是etcd。
+- *声明式配置写入者 /  声明写入者*: 对活动对象进行更新的人员或者软件组件。
+  本主题中提到的写入者对对象的配置文件进行修改，并使用`kubectl apply` 写入这些改动。
+
+## 如何创建对象
+
+使用 `kubectl apply` 创建所有对象（除非已经存在的对象），由指定目录中的配置文件定义。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 译文非常不通顺：Use `kubectl apply` to create all objects, except those that already exist, defined by configuration files in a specified directory:

您可以使用`kubectl apply`来创建指定目录下各个配置文件中定义的所有对象。已经存在的对象不会被创建。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 11]
```diff
@@ -0,0 +1,886 @@
...
+
+```shell
+kubectl apply -f <directory>/
+```
+
+这将在每个对象上设置 `kubectl.kubernetes.io/last-applied-configuration: '{...}'`
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: "这“ => "此操作"

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 12]
```diff
@@ -0,0 +1,886 @@
...
+注释。注释中包含用于创建对象的配置文件的内容。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: annotation一般译为"注解"，以便与comment（注释）相区分。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 13]
```diff
@@ -0,0 +1,886 @@
...
+
+**注**: 添加 `-R` 标签可以递归的处理目录。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: ”flag"这里译为“参数”比较合适，以便与label（标签）相区分。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 14]
```diff
@@ -0,0 +1,886 @@
...
+
+以下是对象配置文件的示例：
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 对象配置文件的示例 => 对象配置文件的一个例子

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 15]
```diff
@@ -0,0 +1,886 @@
...
+
+{% include code.html language="yaml" file="simple_deployment.yaml" ghlink="/cn/docs/tutorials/object-management-kubectl/simple_deployment.yaml" %}
+
+使用 `kubectl apply` 创建对象：
+
+```shell
+kubectl apply -f http://k8s.io/cn/docs/tutorials/object-management-kubectl/simple_deployment.yaml
+```
+
+使用 `kubectl get` 打印实时配置：
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: “实时" ...

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 16]
```diff
@@ -0,0 +1,886 @@
...
+
+```shell
+kubectl get -f http://k8s.io/cn/docs/tutorials/object-management-kubectl/simple_deployment.yaml -o yaml
+```
+
+输出显示 `kubectl.kubernetes.io / last-applied-configuration` 注释已写入实时配置，与配置文件相匹配：
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: "注释"、”实时" ...

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 17]
```diff
@@ -0,0 +1,886 @@
...
+        # ...
+      # ...
+    # ...
+  # ...
+```
+
+## 如何更新对象
+
+您也可以使用`kubectl apply`来更新目录中定义的所有对象，即使这些对象已经存在。
+这种方法完成了以下工作：
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: "工作" => "操作"或者"目标"或者"动作"

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 18]
```diff
@@ -0,0 +1,886 @@
...
+
+1.在实时配置中设置出现在配置文件中的字段。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 请考虑中文的可理解性

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 19]
```diff
@@ -0,0 +1,886 @@
...
+2.清除实时配置中配置文件中删除的字段。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 请考虑中文的可读性

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 20]
```diff
@@ -0,0 +1,886 @@
...
+
+```shell
+kubectl apply -f <directory>/
+```
+
+**注**: 添加 `-R` 标签可以递归的处理目录。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: “标签" => 命令行参数

"directories" 最好译为”多个目录"

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 21]
```diff
@@ -0,0 +1,886 @@
...
+      containers:
+      - image: nginx:1.7.9
+        # ...
+        name: nginx
+        ports:
+        - containerPort: 80
+      # ...
+```
+
+更新“simple_deployment.yaml”配置文件，将镜像从“nginx：1.7.9”更改为“nginx：1.11.9”，
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: `simple_deployment_yaml` `ngnix...`

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 22]
```diff
@@ -0,0 +1,886 @@
...
+并删除“minReadySeconds”字段：
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: `minReadySeconds`

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 23]
```diff
@@ -0,0 +1,886 @@
...
+使用 `kubectl get` 打印实时配置：
+
+```
+kubectl get -f http://k8s.io/cn/docs/tutorials/object-management-kubectl/simple_deployment.yaml -o yaml
+```
+
+输出显示对实时配置的以下更改：
+
+- `replicas` 字段保留 `kubectl scale` 设置的值2。
+  这是可能的，因为它从配置文件中省略了。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 能够做到这一点是因为配置文件中并未给出`replicas`的新值

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 24]
```diff
@@ -0,0 +1,886 @@
...
+        ports:
+        - containerPort: 80
+        # ...
+      # ...
+    # ...
+  # ...
+```
+
+**警告**：
+不支持`kubectl apply`与命令式对象配置命令`create`和`replace`混合使用。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 无主句

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 25]
```diff
@@ -0,0 +1,886 @@
...
+因为`create` 和 `replace`不保留`kubectl.kubernetes.io / last-applied-configuration`字段，
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 斜线前后的空格?

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 26]
```diff
@@ -0,0 +1,886 @@
...
+`kubectl apply`使用这个字段进行更新。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 请重新理解这个警告的含义。译文中完全把原文的含义搞错了。

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 27]
```diff
@@ -0,0 +1,886 @@
...
+
+## 如何删除对象
+
+两种方式删除`kubectl apply`管理的对象。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 这不是一个通顺的完整汉语句子...

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 28]
```diff
@@ -0,0 +1,886 @@
...
+
+```shell
+kubectl delete -f <filename>
+```
+
+### 可选： `kubectl apply -f <directory/> --prune -l your=label`
+
+如果对这条命令不够了解，请不要使用。
+
+**警告：** `kubectl apply --prune` is in alpha, and backwards incompatible
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: ... 未翻译？

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 29]
```diff
@@ -0,0 +1,886 @@
...
+changes might be introduced in subsequent releases.
+
+**警告：** 请慎重使用该命令，防止无意中删除对象。
+
+作为`kubectl delete的`替代方法，您可以使用`kubectl apply`来识别从目录中删除配置文件后要删除的对象。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 请检查这一行

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 30]
```diff
@@ -0,0 +1,886 @@
...
+应用`--prune`查询所有匹配一组标签的对象的API服务器，并尝试将返回的活动对象配置与对象配置文件进行匹配。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: “应用`--prune`查询所有匹配一组标签的对象的API服务器”

请检查

File: cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md [hunk 31]
```diff
@@ -0,0 +1,886 @@
...
+{% comment %}
+TODO(pwittrock): We need to change the behavior to prevent the user from running apply on subdirectories unintentionally.
+{% endcomment %}
+
+```shell
+kubectl apply -f <directory/> --prune -l <labels>
+```
+
+**重要：** 应用修剪应该只应用于包含对象配置文件的根目录。应用于子目录可能会导致对象无意中被删除，
+如果对象由标签选择器指定的标签`-l <labels>`查询返回，并且在子目录中没有的话。
```
Comment (tengqm) [inline cn/docs/tutorials/object-management-kubectl/declarative-object-management-configuration.md]: 请重新组织上面这句话

**Comment by zhangxiaoyu-zidif** [comment]
@Lion-Wei  please fix this PR， thanks.

**Comment by steveperry-53** [comment]
Sorry to be a pest, but we need get through our backlog of PRs. Ping @Lion-Wei.

**Comment by zhangxiaoyu-zidif** [comment]
ping @Lion-Wei

**Comment by zhangxiaoyu-zidif** [comment]
ping @Lion-Wei

**Comment by Lion-Wei** [comment]
Sorry didn't notice, but I don't have enough time for now, how about I close this pr temporarily? I will resubmit it as soon as possible. I am really sorry about that. @chenopis @zhangxiaoyu-zidif @steveperry-53

**Comment by zhangxiaoyu-zidif** [comment]
If you have time to fix the PR, just reopen it. @Lion-Wei

**Comment by Lion-Wei** [comment]
Okay， thx @zhangxiaoyu-zidif

**Pull request closed without merging**